warp10/src/main/java/io/warp10/json/JsonUtils.java

```diff
@@ -0,0 +1,211 @@
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+
+package io.warp10.json;
+
+import com.fasterxml.jackson.core.JsonFactoryBuilder;
+import com.fasterxml.jackson.core.JsonGenerator;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.core.StreamWriteFeature;
+import com.fasterxml.jackson.core.json.JsonReadFeature;
+import com.fasterxml.jackson.core.json.JsonWriteFeature;
+import com.fasterxml.jackson.databind.BeanDescription;
+import com.fasterxml.jackson.databind.JsonSerializer;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationConfig;
+import com.fasterxml.jackson.databind.SerializerProvider;
+import com.fasterxml.jackson.databind.module.SimpleModule;
+import com.fasterxml.jackson.databind.ser.BeanSerializer;
+import com.fasterxml.jackson.databind.ser.BeanSerializerModifier;
+import com.fasterxml.jackson.databind.ser.impl.UnknownSerializer;
+import io.warp10.continuum.gts.GTSEncoder;
+import io.warp10.continuum.gts.GeoTimeSerie;
+import io.warp10.continuum.store.thrift.data.Metadata;
+import io.warp10.script.NamedWarpScriptFunction;
+import io.warp10.script.WarpScriptStack;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.io.Writer;
+import java.util.ArrayList;
+import java.util.List;
+
+public class JsonUtils {
+
+  /**
+   * A serializer for null keys.
+   * Outputs "null" because most javascript engines coerce null to "null" when using it as a key.
+   */
+  private static class NullKeySerializer extends JsonSerializer<Object> {
+    @Override
+    public void serialize(Object value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
+      gen.writeFieldName("null");
+    }
+  }
+
+  /**
+   * Used to swap UnknownSerializer and BeanSerializer for CustomEncodersSerializer.
+   */
+  public static class NotSerializedToCustomSerializedModifier extends BeanSerializerModifier {
+    @Override
+    public JsonSerializer<?> modifySerializer(SerializationConfig config, BeanDescription beanDesc, JsonSerializer<?> serializer) {
+      if (serializer instanceof UnknownSerializer || serializer instanceof BeanSerializer) {
+        return customEncodersSerializer;
+      } else {
+        return serializer;
+      }
+    }
+  }
+
+  public static class CustomEncodersSerializer extends JsonSerializer<Object> {
+    @Override
+    public void serialize(Object value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
+      if (null != encoders && !encoders.isEmpty()) {
+        StringBuilder sb = new StringBuilder();
+        boolean encoded = false;
+        for (JsonEncoder encoder: encoders) {
+          encoded = encoder.addElement(sb, value);
+          if (encoded) {
+            break;
+          }
+        }
+        if (encoded) {
+          gen.writeString(sb.toString());
```

Comment:
Won't that surround sb's content with double quotes and escape it? This might not be what is expected.